Integrations/integration-CrowdStrikeHost.yml

```diff
@@ -0,0 +1,647 @@
+script:
+  script: |
+    var serverUrl = params.url;
+    if (serverUrl[serverUrl.length - 1] !== '/') {
+        serverUrl += '/';
+    }
+
+    var doReq = function(method, path, query, body) {
+        var result = http(
+            serverUrl + path + encodeToURLQuery(query),
+            {
+                Headers: {
+                    'Content-Type': ['application/json'],
+                    'Accept': ['application/json']
+                },
+                Username: params.id,
+                Password: params.key,
+                Method: method,
+                Body: body ? JSON.stringify(body) : ''
+            },
+            params.insecure,
+            params.useproxy
+        );
+        if (result.StatusCode < 200 && result.StatusCode > 299) {
+            throw 'Failed to perform request ' + path + ', request status code: ' + result.StatusCode + ', body: ' + result.Body;
+        }
+        if (result.Body === '') {
+            throw 'No content recieved.';
+        }
+        var obj;
+        try {
+            obj = JSON.parse(result.Body);
+        } catch (ex) {
+            throw 'Error parsing reply - ' + result.Body + ' - ' + ex;
+        }
+        if (obj.errors && obj.errors.length > 0) {
+            throw obj.errors.join(', ');
+        }
+        return {body: result.Body, obj: obj, statusCode: result.StatusCode};
+    };
+
+    var dateToEpoch = function(d) {
+        return d ? new Date(d).getTime() : null;
+    };
+
+    var uploadIOC = function() {
+        var res = doReq('POST', 'indicators/entities/iocs/v1', {}, [args]);
+        return {Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: 'IOC [' + args.value + '] uploaded'};
+    };
+
+    var getIOC = function() {
+        var res = doReq('GET', 'indicators/entities/iocs/v1', {ids: args.type + ':' + args.value});
+        var md = '### Falcon Host custom IOC retrieval\n';
+        if (res.obj.resources) {
+            for (var i=0; i<res.obj.resources.length; i++) {
+                md += objToList(res.obj.resources[i], res.obj.resources[i].value) + '\n';
+            }
+        } else {
+            md += 'No result found';
+        }
+        return {Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: md};
+    };
+
+    // add args to a if exists
+    var add = function(a, k, f) {
+        if (args[k]) {
+            var parts = argToList(args[k]);
+            for (var i=0; i<parts.length; i++) {
+                parts[i] = parts[i].trim();
+            }
+            a[k] = f ? f(args[k]) : parts.length > 1 ? parts : parts[0];
+        }
+    };
+
+    var updateIOC = function() {
+        var a = {};
+        add(a, 'policy');
+        add(a, 'share_level');
+        add(a, 'expiration_days');
+        add(a, 'source');
+        add(a, 'description');
+        var res = doReq('PATCH', 'indicators/entities/iocs/v1', {ids: args.type + ':' + args.value}, a);
```

Comment:
In many cases I needed to change the name and even format. Not in this case but wanted to keep the same style. Also, easier to add what's known than to delete everything else...